upgrade/steps.go

```diff
@@ -574,27 +575,34 @@ func MajorVersionBump(ctx Context, goMod *GoMod, target *UpstreamUpgradeTarget,
 			return nextMajorVersion, nil
 		}),
 		replaceInFile("Update PROVIDER_PATH", "Makefile",
-			"PROVIDER_PATH := {}").In(&repo.root),
+			"PROVIDER_PATH := {}",
+		).In(&repo.root),
 		replaceInFile("Update -X Version", ".goreleaser.yml",
-			"github.com/pulumi/"+name+"/{}/pkg").In(&repo.root),
+			"github.com/pulumi/"+name+"/{}/pkg",
+		).In(&repo.root),
 		replaceInFile("Update -X Version", ".goreleaser.prerelease.yml",
-			"github.com/pulumi/"+name+"/{}/pkg").In(&repo.root),
-		replaceInFile("Update Go Module", "go.mod",
-			"module github.com/pulumi/"+name+"/{}").In(repo.providerDir()),
-		step.F("Update sdk/go.mod", func() (string, error) {
-			path := "sdk/go.mod"
-			f, err := os.ReadFile(path)
-			if err != nil {
-				return "", err
-			}
-			mod := fmt.Sprintf("github.com/%s/%s/sdk", repo.org, repo.name)
-			var prev string
-			if m := repo.currentVersion.Major(); m > 1 {
-				prev = fmt.Sprintf("/%d", m)
+			"github.com/pulumi/"+name+"/{}/pkg",
+		).In(&repo.root),
+		UpdateFile("Update Go Module in provider/go.mod", "go.mod",
+			func(src []byte) ([]byte, error) {
+				base := "module github.com/pulumi/" + name + "/provider"
+				old := base
+				if repo.currentVersion.Major() > 1 {
+					old += fmt.Sprintf("/v%d", repo.currentVersion.Major())
+				}
+				new := base + fmt.Sprintf("/v%d", repo.currentVersion.IncMajor().Major())
+
+				return bytes.ReplaceAll(src, []byte(old), []byte(new)), nil
```

Comment:
I think I'm missing something; how is this different than the 
```
replaceInFile("Update Go Module", "go.mod",
			"module github.com/pulumi/"+name+"/{}").In(repo.providerDir()),
```
step it replaced?

Reply:
I exploded it when I was debugging. Its' not. https://github.com/pulumi/upgrade-provider/pull/129/commits/b6a42f75badf46f22f7bbb34e31bac3b31025b31 has compressed it back down.